src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -232,9 +233,13 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
                 checkpoint_dir=checkpoint_dir, early_stopping_data_name=early_stopping_data_name,
                 early_stopping_metric=early_stopping_metric, early_stopping_rounds=early_stopping_rounds,
                 save_model_on_termination=save_model_on_termination, is_master=is_master)
-            add_debugging(callbacks=callbacks, hyperparameters=train_cfg, train_dmatrix=train_dmatrix,
-                          val_dmatrix=val_dmatrix)
-
+            if XGB_VERSION < "1.3.0":
+                # when xgboost >= 1.3.0, it doesn't allow callback which are not instances 
+                # of callback.TrainingCallback
+                # ref : https://github.com/dmlc/xgboost/blob/v1.5.2/python-package/xgboost/training.py#L62
+                #       https://xgboost.readthedocs.io/en/latest/python/callbacks.html
```

Comment:
Per the code and doc links, can we solve this issue just by wrapping the offending callbacks as TrainingCallbacks? https://xgboost.readthedocs.io/en/latest/python/callbacks.html#defining-your-own-callback

That way we wouldn't need this check at all.

Reply:
it's from sagemaker debugger. https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/xgboost/hook.py#L36

Reply:
oh, wait , you mean wrap the hook by TrainingCallbacks? it should work but I think this sounds more like the work for debugger team. they only supports xgboost 0.90-2, 1.0-1 from readme. If we really need it, we can implemement our own callback instead.

Reply:
Yeah, exactly. I agree that ideally this should be a change made in smedebug, but I think it's worth evaluating the level of effort to convert the old API to the new one. This is the new API spec: https://xgboost.readthedocs.io/en/stable/python/python_api.html#callback-api

We could define a module similar to this: https://github.com/aws/sagemaker-xgboost-container/tree/b42040bf32cb8b59c444c0f92c3bef44c29ca902/src/sagemaker_xgboost_container/dmlc_patch that defines `class SmeDebugHook(xgb.callback.TrainingCallback)`. I'm not even sure we need to override the methods: https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/callback.py#L36 in order to get it to work, but that should be tested.

Let's open an issue with the SageMaker Debugger team https://github.com/awslabs/sagemaker-debugger/issues so they're aware of this issue, in any case.